### PR TITLE
Add support for destroy_data and update_data to be sent

### PR DESCRIPTION
**Pull request by jgrumboe**
This PR is inspired by #177 and it adds one feature:
- destroy_data => Allows specifying a 'data/body' to send along with destroy/delete requests.

It could solve #174 I think.

### CODE REVIEW

**Comment by jgrumboe** [comment]
@DRuggeri Is this PR ok? Is the test sufficient?

If it's fine I could also "extend" this PR to solve #124 and add `update_data` in the same way.

**Comment by m-garrido** [comment]
It would be awesome to have the same feature for update and delete.

**Comment by jgrumboe** [comment]
@m-garrido @melt-melt
I also added support for `update_data`.
@DRuggeri can you have a look at this?

**Comment by jgrumboe** [comment]
@DRuggeri I needed to update the GH release action also following this comment https://github.com/hashicorp/ghaction-import-gpg/issues/11#issuecomment-1185107410
Otherwise I wasn't able to publish the provider interim to my personal TF registry account.

I can remove the commit before merging.

**Comment by jgrumboe** [comment]
@m-garrido @melt-melt
I published my changes interim to TF registry under https://registry.terraform.io/providers/jgrumboe/restapi/1.18.0-dev
You could try out `update_data` and `destroy_data` for tests, would like to remove the provider from my account in the TF registry after this PR is merged.

**Review by arun-a-nayagam — COMMENTED** [review]
I have tested the update_data changes in an actual project and can confirm that it works fine.

**Comment by jgrumboe** [comment]
@DRuggeri can you help me here to get that merged? If something is missing like tests or docs, just tell me. Thanks in advance 🙏

**Comment by jgrumboe** [comment]
@douernesto @Martina-May 
Could you probably help in getting this PR merged? Unfortunately there's no feedback yet from @DRuggeri 😞

**Comment by Sh1ftry** [comment]
Hello, I am also waiting for this feature. Would be nice to have it released soon.

**Review by DRuggeri — CHANGES_REQUESTED** [review]
This looks good, though I am concerned about some of the changes to the release workflow. Can you please address those?

**Comment by DRuggeri** [comment]
Many apologies for the delay responding to this PR. This seems to make a lot of sense and I've added a quick few notes. If you can address those, we can go ahead and merge and cut a new release with this!

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -1,49 +1,52 @@
...
         name: Set up Go
-        uses: actions/setup-go@v2
+        uses: actions/setup-go@v3
         with:
-          go-version: 1.17
+          go-version-file: 'go.mod'
+          cache: true
       -
         name: Import GPG key
+        uses: crazy-max/ghaction-import-gpg@v5
```
[the remark this answers is not in the document]
Comment (jgrumboe) [reply .github/workflows/release.yaml:37]: This change is encouraged by one of hashicorp gh actions maintainer: https://github.com/hashicorp/ghaction-import-gpg/issues/11#issuecomment-1185107410
(also commented it here some time ago)

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -1,49 +1,52 @@
...
+          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.PASSPHRASE }}
       -
         name: Run GoReleaser
-        uses: goreleaser/goreleaser-action@v2.8.1
+        uses: goreleaser/goreleaser-action@v3.0.0
         with:
           version: latest
-          args: release --parallelism 2 --rm-dist --timeout 1h --release-notes .release_info.md
+          args: release --rm-dist
```
[the remark this answers is not in the document]
Comment (jgrumboe) [reply .github/workflows/release.yaml]: Fixed in 225bd5f

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -1,49 +1,52 @@
 # This GitHub action can publish assets for release when a tag is created.
 # Currently its setup to run on any tag that matches the pattern "v*" (ie. v0.1.0).
 #
-# This uses an action (hashicorp/ghaction-import-gpg) that assumes you set your
+# This uses an action (hashicorp/ghaction-import-gpg) that assumes you set your
```
[the remark this answers is not in the document]
Comment (jgrumboe) [reply .github/workflows/release.yaml]: Fixed in 225bd5f

File: restapi/api_object.go [hunk 4]
```diff
@@ -50,6 +52,8 @@ type APIObject struct {
 
 	/* Set internally */
 	data        map[string]interface{} /* Data as managed by the user */
+	updateData  map[string]interface{} /* Update data as managed by the user */
+	destroyData map[string]interface{} /* Update data as managed by the user */
```
[the remark this answers is not in the document]
Comment (jgrumboe) [reply restapi/api_object.go]: Fixed in cb4ab82

File: docs/resources/object.md [hunk 5]
```diff
@@ -34,6 +35,7 @@ description: |-
 - **read_path** (String, Optional) Defaults to `path/{id}`. The API path that represents where to READ (GET) objects of this type on the API server. The string `{id}` will be replaced with the terraform ID of the object.
 - **read_search** (Map of String, Optional) Custom search for `read_path`. This map will take `search_key`, `search_value`, `results_key` and `query_string` (see datasource config documentation)
 - **update_method** (String, Optional) Defaults to `update_method` set on the provider. Allows per-resource override of `update_method` (see `update_method` provider config documentation)
+- **update_data** (String, Optional) Valid JSON data that this provider will send to the API server on UPDATE (PUT) operations. Allows override of `data` on UPDATE (PUT) operations.
```
[the remark this answers is not in the document]
Comment (jgrumboe) [reply docs/resources/object.md]: Fixed in cb4ab82

File: docs/resources/object.md [hunk 6]
```diff
@@ -24,6 +24,7 @@ description: |-
 - **create_path** (String, Optional) Defaults to `path`. The API path that represents where to CREATE (POST) objects of this type on the API server. The string `{id}` will be replaced with the terraform ID of the object if the data contains the `id_attribute`.
 - **debug** (Boolean, Optional) Whether to emit verbose debug output while working with the API object on the server.
 - **destroy_method** (String, Optional) Defaults to `destroy_method` set on the provider. Allows per-resource override of `destroy_method` (see `destroy_method` provider config documentation)
+- **destroy_data** (String, Optional) Valid JSON data that this provider will send to the API server on DESTROY (DELETE) operations.
```
[the remark this answers is not in the document]
Comment (jgrumboe) [reply docs/resources/object.md]: Fixed in cb4ab82

**Comment by jgrumboe** [comment]
@DRuggeri Thanks for the feedback. I addressed it.

File: .github/workflows/release.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (DRuggeri) [reply .github/workflows/release.yaml:37]: Got it - thanks!

**Review by DRuggeri — APPROVED** [review]

**Pull request merged**